packages/client/src/components/ModuleTree/Module/Module.tsx

```diff
@@ -1,25 +1,104 @@
 import React from 'react';
-import { Module as EModule } from '@codement/api';
-import classnames from 'classnames';
-import { Icon } from '@codement/ui';
-import Hex from './hex.svg';
-import styles from './Module.module.css';
+// import classnames from 'classnames';
+
+import { Module as EModule, UserModule, UserConcept } from '@codement/api';
+import { theme } from '@codement/ui';
+
+import { ModuleIcon } from './Icons/Icon';
 
 export interface ModuleProps {
-  module: EModule
+  module: EModule,
+  myModule?: UserModule,
+  myConcepts?: UserConcept[]
 }
 
 export const Module: React.FC<ModuleProps> = ({
-  module
-}) => <div className={styles.module}>
-  <div className="relative">
-    <Hex
-      className={classnames('h-12 stroke-2', {
-        // TODO: Add completed to module
-        completed: false
-      })}
-    />
-    <Icon icon="fire" size={4} />
-  </div>
-  <span className="text-xs">{module.name}</span>
-</div>;
+  module,
+  myModule,
+  myConcepts
+}) => {
+  const isCompleted = myModule?.completedAt;
+  const isOpen = myModule && !myModule.completedAt;
+  const isLocked = !myModule;
+
+  return <div style={{
+    textAlign: 'center',
+    marginBottom: '1.5rem',
+    color: theme.colors.primary['500'] }}
+  >
+
+    <div style={{
+      position: 'relative',
+      display: 'flex',
+      height: '3rem',
+      width: '3rem',
+      marginRight: 'auto',
+      marginLeft: 'auto' }}
+    >
+
+      <ModuleIcon
+        icon={module.type === 'lesson' ? 'hexagon' : 'octagon'}
+        position="absolute"
+        size={3}
+        // eslint-disable-next-line no-nested-ternary
+        fill={isCompleted ? theme.colors.secondary['500'] : isOpen ? theme.colors.primary['400'] : theme.colors.white}
+        stroke={isLocked ? theme.colors.grey['500'] : undefined}
+        left="50%"
+        top="50%"
+        transform="translate(-50%, -50%)"
+      />
+
+      { module.type !== 'assignment'
+      && <ModuleIcon
+        icon="circle"
+        position="absolute"
+        size="small"
+        // eslint-disable-next-line no-nested-ternary
```

Comment:
Can we disable this globally @tristanMatthias or it's actually a good practice to not have nested ternary operators